flink/src/main/java/org/apache/iceberg/flink/source/FlinkInputFormat.java

```diff
@@ -91,11 +102,77 @@ public void configure(Configuration parameters) {
 
   @Override
   public void open(FlinkInputSplit split) {
+    boolean enableVectorizedRead = readableConfig.get(FlinkTableOptions.ENABLE_VECTORIZED_READ);
+
+    if (enableVectorizedRead) {
+      if (useOrcVectorizedRead()) {
```

Comment:
There are other required requisition so that we could apply vectorized read:
1.   All those files from the `CombinedScanTask` are data files,  if there is a delete file, the current deletions apply process are compared row by row which will disable the vectorized read actually. 
2.  All the files from `CombinedScanTask` must be ORC files.
3.  All the columns to read should all be primitives,  that means all the byte width should be the same size.

Reply:
> 3\. All the columns to read should all be primitives,  that means all the byte width should be the same size

We don't support `Map`, `Row`, `Array` data type?